django_elasticsearch_dsl/management/commands/search_index.py

```diff
@@ -123,29 +152,139 @@ def _populate(self, models, options):
+            # Elasticsearch will return an error if an index already
+            # exists with the desired alias name. Therefore, we need to
+            # delete that index.
+            delete_existing_index = True
+            alias_actions.append({"remove_index": {"index": alias}})
+
+        old_indices = []
+        alias_delete_actions = []
+        if alias_exists:
+            # Elasticsearch will return an error if we search for
+            # indices by alias but the alias doesn't exist. Therefore,
+            # we want to be sure the alias exists.
+            old_indices = self._get_alias_indices(alias)
+            alias_actions.append(
+                {"remove": {"alias": alias, "indices": old_indices}}
+            )
+            alias_delete_actions = [
+                {"remove_index": {"index": index}} for index in old_indices
+            ]
+
+        self.es_conn.indices.update_aliases({"actions": alias_actions})
+        if delete_existing_index:
+             self.stdout.write("Deleted index '{}'".format(alias))
+
+        self.stdout.write(
+            "Added alias '{}' to index '{}'".format(alias, new_index)
+        )
+
+        if old_indices:
+            for index in old_indices:
+                self.stdout.write(
+                    "Removed alias '{}' from index '{}'".format(alias, index)
+                )
+
+            if alias_delete_actions and not options['use_alias_keep_index']:
+                self.es_conn.indices.update_aliases(
+                    {"actions": alias_delete_actions}
+                )
+                for index in old_indices:
+                    self.stdout.write("Deleted index '{}'".format(index))
+
+    def _rebuild(self, models, aliases, options):
+        if (not options['use_alias']
+            and not self._delete(models, aliases, options)):
             return
 
-        self._create(models, options)
+        if options['use_alias']:
+            alias_index_pairs = []
+            index_suffix = "-" + datetime.now().strftime("%Y%m%d%H%M%S%f")
+            for index in registry.get_indices(models):
+                # The alias takes the original index name value. The
+                # index name sent to Elasticsearch will be the alias
+                # plus the suffix from above. In addition, the index
+                # name needs to be limited to 255 characters, of which
+                # 21 will always be taken by the suffix, leaving 234
+                # characters from the original index name value.
+                new_index = index._name[:234] + index_suffix
+                alias_index_pairs.append(
+                    {'alias': index._name, 'index': new_index}
+                )
+                index._name = new_index
+        else:
+            for index in registry.get_indices(models):
+                alias_exists = index._name in aliases
+                if alias_exists:
+                    self._delete_alias_indices(index._name)
```

Comment:
is it possible to move this code block to the `_delete()` method?

Reply:
Good call on this one! I updated it in 5b53d5a.